### PR TITLE
Fix #3323: Revert back to standard share sheet elements based on URL

### LINKED ISSUE #3323: Add ability to save PDF opened in Brave via share sheet menu
### Description: 
Safari has this option:

IMG_C9DABC041D5F-1

### Steps to Reproduce 
  1. Open a pdf
  2. Open share menu

**Actual result:** 

**Expected result:**

**Reproduces how often:** [Easily reproduced, Intermittent Issue]

**Brave Version:** 

**Device details:** 

**Website problems only:**
- did you check with Brave Shields down?
- did you check in Safari/Firefox (WkWebView-based browsers)? 

### Additional Information
Reported here
https://community.brave.com/t/how-to-download-a-pdf-file-on-iphone/207624

### PR DESCRIPTION
**Pull request by cuba**
## Summary of Changes

This pull request fixes #3323

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`

## Test Plan:
STR included in ticket #3323, however some additional scenarios need to be tested, including:
* Create PDF share sheet for reader mode
* Create PDF share sheet for regular mode
* Share site on reader mode
* Share site on regular mode
* Test sharing brave shield stats
* Test sharing large PDF
* Test emailing pdf
* Any other business scenarios I may not be familiar with.
* STR in #2961

## Screenshots:

IMG_6649031F0873-1
IMG_1F5C0904A356-1
IMG_5FEA3404B4CA-1

Note: The following scenario doesn't support "Save to file"
IMG_CCA1137EE40F-1

## Reviewer Checklist:

- [x] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [x] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [x] Adequate unit test coverage exists to prevent regressions.
- [x] Adequate test plan exists for QA to validate (if applicable).
- [x] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: Client/Frontend/Share/ShareExtensionHelper.swift [hunk 1]
```diff
@@ -109,84 +92,14 @@ class ShareExtensionHelper: NSObject {
                         completion(url, nil)
                     } catch {
                         completion(nil, error)
-                        log.error("Failed to write PDF to disk: \(error)")
+                        Logger.browserLogger.error("Failed to write PDF to disk: \(error)")
```
Comment (iccub) [inline Client/Frontend/Share/ShareExtensionHelper.swift]: No need to change those logger methods, this is code convention we use across our app.

File: Client/Frontend/Share/ShareExtensionHelper.swift [hunk 2]
```diff
@@ -60,36 +51,28 @@ class ShareExtensionHelper: NSObject {
...
         }
         
         return activityViewController
     }
-    
-    func writeWebPagePDFDataToURL(_ completion: @escaping (URL?, Error?) -> Void) {
+
+    // TODO: @JS Figure of if this is ever used
+    /// Function that writes the current webpage data to a PDF
+    private static func writeWebPagePDFDataToURL(selectedTab: Tab, completion: @escaping (URL?, Error?) -> Void) {
```
Comment (iccub) [inline Client/Frontend/Share/ShareExtensionHelper.swift]: Let's wait for @soner-yuksel review, then we can remove this method, i did not see it being called anywhere

Comment (soner-yuksel) [reply Client/Frontend/Share/ShareExtensionHelper.swift]: Yes we can absolutely remove this method.

**Review by soner-yuksel — APPROVED** [review]
Good work!

**Pull request merged**